docs/deployer/terraform.md

```diff
@@ -0,0 +1,168 @@
+# Terraform Deployer
+
+The Terraform deployer is a controller that reconciles
+DeployItems of type `landscaper.gardener.cloud/terraform`.
+
+It manages infrastructure resources using terraform.
+This means running the `apply` and `destroy` terraform commands.
+
+To run these commands, it relies on the Gardener component
+[Terraformer](https://github.com/gardener/terraformer)
+
+### Configuration
+
+This sections describes the provider specific configuration
+
+```yaml
+apiVersion: landscaper.gardener.cloud/v1alpha1
+kind: DeployItem
+metadata:
+  name: s3-bucket
+spec:
+  type: landscaper.gardener.cloud/terraform
+
+  target: # has to be of type landscaper.gardener.cloud/kubernetes-cluster
+    name: my-cluster
+    namespace: test
+
+  config:
+    apiVersion: terraform.deployer.landscaper.gardener.cloud/v1alpha1
+    kind: ProviderConfiguration
+
+    # Use a specific image for the Terraformer
+    # optional
+    terraformerImage: eu.gcr.io/gardener-project/gardener/terraformer:v1.5.0
+
+    # base64 encoded kubeconfig pointing to the cluster to install the chart
+    # optional if Target is defined.
+    kubeconfig: <redacted>
+
+    # Namespace where the terraformer pod will run and where it will
+    # store the main.tf, variables.tf as ConfigMap and terraform.tfvars as Secret.
+    # The secrets below must exist in this namespace.
+    # option - default to default namespace.
+    namespace: default
+
+    # Secrets names containing environment variables which will be
+    # added to the Terraformer pod.
+    # optional
+    envSecrets:
+      - aws          # infrastructure secret with AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY
+      - application  # application secret with TF_VAR_bucket_b
```

Comment:
The idea here was to used pre-created secrets to avoid storing credentials in the de `DeployItem` and makes the secret reusable.

It differs a bit from the `terraformer` implementation where the credentials are stored in `terraform.tfvars`, if it's fine having the credentials there, this field would not be required, what do you say ?

Reply:
I think for now this is the only way how we can pass secrets from the landscaper to the deploy items.
We need to solve that issue in general So its currently fine for you to assume that the secrets are stored in the deploy item